src/aap_eda/core/models/activation.py

```diff
@@ -74,6 +82,42 @@ class Meta:
     status_updated_at = models.DateTimeField(null=True)
     status_message = models.TextField(null=True, default=None)
 
+    def save(self, *args, **kwargs):
+        # when creating
+        if self._state.adding:
+            if self.status_message is None:
+                self._set_status_message()
+        else:
+            if not bool(kwargs):
+                raise UpdateFieldsRequiredError(
+                    "update_fields is required to use when saving "
+                    "due to race conditions"
+                )
+            # when updating without status
+            elif (
+                "status_message" in kwargs["update_fields"]
+                and "status" not in kwargs["update_fields"]
+            ):
+                raise StatusMessageRequiredError(
```

Comment:
@hsong-rh This should be StatusRequiredError instead of StatusMessageRequiredError. Because we are saying that status is mandatory, status_message is optional and status_message can't be set by itself. Maybe the error message should say "status_message cannot be set by itself, it requires status and status_message together".